### ISSUE TITLE
ERROR: feedbackgenerator::apply_sorting(): Argument #2 ($selectedscaleid) must be of type int, null given

### ISSUE THREAD

**Issue by ralferlebach** [Wunderbyte-GmbH/moodle-local_catquiz#744]
Teststrategie umgestellt, nächster Versuch lieferte folgenden Fehler:

![Image](https://github.com/user-attachments/assets/3af1c609-4c8a-415b-b1a6-e2c8c64506b2)

Fehler: local_catquiz\teststrategy\feedbackgenerator::apply_sorting(): Argument #2 ($selectedscaleid) must be of type int, null given, called in [dirroot]/local/catquiz/classes/teststrategy/feedbackgenerator.php on line 419

[Weitere Informationen über diesen Fehler](https://docs.moodle.org/401/de/error/moodle/generalexceptionmessage)
Debug-Info: 
Error code: generalexceptionmessage
Stack trace: 

    line 449 of /local/catquiz/classes/teststrategy/feedbackgenerator.php: TypeError thrown
    line 419 of /local/catquiz/classes/teststrategy/feedbackgenerator.php: call to local_catquiz\teststrategy\feedbackgenerator->apply_sorting()
    line 332 of /local/catquiz/classes/teststrategy/feedbackgenerator/comparetotestaverage.php: call to local_catquiz\teststrategy\feedbackgenerator->get_restructured_abilities()
    line 170 of /local/catquiz/classes/teststrategy/feedbackgenerator.php: call to local_catquiz\teststrategy\feedbackgenerator\comparetotestaverage->load_data()
    line 335 of /local/catquiz/classes/output/attemptfeedback.php: call to local_catquiz\teststrategy\feedbackgenerator->update_data()
    line 190 of /local/catquiz/classes/output/attemptfeedback.php: call to local_catquiz\output\attemptfeedback->load_data_from_generators()
    line 219 of /local/catquiz/classes/teststrategy/strategy.php: call to local_catquiz\output\attemptfeedback->update_feedbackdata()
    line 165 of /local/catquiz/classes/teststrategy/strategy.php: call to local_catquiz\teststrategy\strategy->update_attemptfeedback()
    line 890 of /local/catquiz/classes/catquiz_handler.php: call to local_catquiz\teststrategy\strategy->return_next_testitem()
    line 87 of /mod/adaptivequiz/catmodel/catquiz/classes/local/catmodel/itemadministration/catquiz_item_administration.php: call to local_catquiz\catquiz_handler::fetch_question_id()
    line 106 of /mod/adaptivequiz/classes/local/adaptive_quiz_session.php: call to adaptivequizcatmodel_catquiz\local\catmodel\itemadministration\catquiz_item_administration->evaluate_ability_to_administer_next_item()
    line 141 of /mod/adaptivequiz/attempt.php: call to mod_adaptivequiz\local\adaptive_quiz_session->administer_next_item_or_stop()

**Comment by ralferlebach** [comment]
Kam nun schon wieder vor, diesmal bei "all subscales".

**Comment by ralferlebach** [comment]
Wir haben gerade im Einsatz massive Probleme mit diesem Fehler. Versuche, die diesen Fehler werfen, können nicht fortgesetzt werden.

Daher

**Comment by davidszkiba** [comment]
@ralferlebach kann es sein, dass in den Quiz Settings die Hauptskala nicht selektiert ist? Ich habe diesen Fehler reproduzieren können, wenn das so ist. Die commits in diesem branch sollten das beheben. Ich warte noch darauf, dass die github actions grün sind, und merge dann

**Comment by ralferlebach** [comment]
Ja, cool. Exakt so ist es: Die Hauptskala ist bzgl. Reportieren abgewählt.

**Comment by ralferlebach** [comment]
Das Problem trat jedoch unregelmäßig auf: Manche TN hatten diesen Fehler nach 3 Fragen, andere nach 12 und andere überhaupt nicht. Alle TN nutzten ALiSe zu selben Zeit mit den selben Einstellungen.

**Comment by davidszkiba** [comment]
Hm ok.
Ich hab gerade einen PR erstellt. Beim pushen waren die github actions leider rot. Es scheinen jetzt ab und zu manche runs rot zu sein und andere grün. Wenn bei der action für den PR die meisten runs grün sind, denke ich, dass wir mergen können.
